decisions/websre/0005-web-sre-kubernetes-secrets-pattern.md

```diff
@@ -0,0 +1,40 @@
+# 1. web-sre-kubernetes-secrets-pattern
+
+Date: 2021-06-09
+
+## Status
+
+Accepted
+
+## Context
+
+Secrets are inconsistently managed across our various clusters.  This makes it hard to trace where secrets come from and update them.  It also increases our exposure to security issues, as our each tool used can become compromised independently.  Current known secret management includes git-crypt repos, with sops, that apply kubernetes secrets direct to the cluster; sealed secrets; and external secrets pointing at cloud provider secret storage.  There are a few other methods that treat things that should be secrets incorrectly. For example, baking the 'secrets' into the docker image directly. We will treat those situations with more urgency, viewing them as explicitly a bad state to be in.
```

Comment:
> baking the secrets into the docker image directly

This method isn't listed below in the "we don't like these ways of doing it" list, for whatever it's worth. No objections from me, though.

> viewing them as explicitly a bad state to be in.

Is that a statement that we have begun to do so today? Or is that an implied outcome of this ADR's acceptance? It's not listed in the goal below, so I wanted to check.